File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.stream.IntStream;

class ResistorColor {
	static final String[] COLORS = { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey",
			"white" };

	int colorCode(String color) {
		return IntStream.range(0, COLORS.length).filter(i -> COLORS[i].equals(color)).findAny().getAsInt();
	}

	String[] colors() {
		return COLORS;
	}
}

File: csharp/resistor-color/ResistorColor.cs
using System;

public static class ResistorColor
{
    static readonly string[] COLORS =
    {
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white"
    };

    public static int ColorCode(string color) => Array.IndexOf(COLORS, color);

    public static string[] Colors() => COLORS;
}
